pytorch_lightning/loops/optimizer/optimizer_loop.py

```diff
@@ -20,19 +21,143 @@
 
 from pytorch_lightning.core.optimizer import LightningOptimizer
 from pytorch_lightning.loops import Loop
-from pytorch_lightning.loops.closure import Closure, ClosureResult
+from pytorch_lightning.loops.closure import AbstractClosure, OutputResult
 from pytorch_lightning.loops.utilities import (
     _block_parallel_sync_behavior,
     _build_training_step_kwargs,
-    _check_training_step_output,
     _extract_hiddens,
     check_finite_loss,
 )
+from pytorch_lightning.profiler import BaseProfiler, PassThroughProfiler
 from pytorch_lightning.trainer.progress import OptimizationProgress
 from pytorch_lightning.utilities import AMPType, DeviceType, grad_norm
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from pytorch_lightning.utilities.finite_checks import detect_nan_parameters
 from pytorch_lightning.utilities.imports import _TPU_AVAILABLE
+from pytorch_lightning.utilities.memory import recursive_detach
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+from pytorch_lightning.utilities.warnings import WarningCache
+
+
+@dataclass
+class ClosureResult(OutputResult):
```

Comment:
`ClosureResult` and `Closure` are now defined alongside the `OptimizerLoop` aka `AutomaticLoop`.

We could also create a folder for it

```python
optimization/
    manual/
        loop.py
        closure.py
    automatic/
        loop.py
        closure.py
```

Reply:
- I would define the specific result definitions directly in the loop files. 
- closure.py can't really go with manual_loop

Reply:
> closure.py can't really go with manual_loop

oops yes, meant to write `result.py`

> I would define the specific result definitions directly in the loop files.

So keep it as it is?

Reply:
yes, I think that would serve us the best